nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/TrainingConfig.java

```diff
@@ -348,17 +357,23 @@ public Builder skipBuilderValidation(boolean skip){
             return this;
         }
 
+        public Builder minimize(String... lossVariables){
+            this.lossVariables = Arrays.asList(lossVariables);
+            return this;
+        }
+
         public TrainingConfig build(){
             if(!skipValidation) {
                 Preconditions.checkState(updater != null, "Updater (optimizer) must not be null. Use updater(IUpdater) to set an updater");
                 Preconditions.checkState(dataSetFeatureMapping != null, "No DataSet feature mapping has been provided. A " +
-                        "mapping between DataSet array positions and variables/placeholders must be provided - use  dateSetFeatureMapping(...) to set this");
-                Preconditions.checkState(dataSetLabelMapping != null, "No DataSet label mapping has been provided. A " +
-                        "mapping between DataSet array positions and variables/placeholders must be provided - use  dateSetLabelMapping(...) to set this");
+                        "mapping between DataSet array positions and variables/placeholders must be provided - use dateSetFeatureMapping(...) to set this");
+                Preconditions.checkState(markLabelsUnused || dataSetLabelMapping != null, "No DataSet label mapping has been provided. A " +
+                        "mapping between DataSet array positions and variables/placeholders must be provided - use dateSetLabelMapping(...) to set this," +
```

Comment:
typo: dateSetFeatureMapping and dateSetLabelMapping should be dataSetFeatureMapping and dataSetLabelMapping